packages/@o3r/core/schematics/shared/presets/helpers.ts

```diff
@@ -14,18 +15,39 @@ import { AddDevInstall } from '@o3r/schematics';
 export function defaultPresetRuleFactory(moduleToInstall: string[]) {
   const corePackageJsonContent = JSON.parse(fs.readFileSync(path.resolve(__dirname, '..', '..', '..', 'package.json'), { encoding: 'utf-8' })) as PackageJson;
   const o3rCoreVersion = corePackageJsonContent.version ? `@${corePackageJsonContent.version}` : '';
+  const getInstalledVersion = async (packageName: string) => {
+    try {
+      return (await import(`${packageName}/package.json`)).version;
+    } catch (e) {
+      return;
+    }
+  };
 
   return (options: PresetOptions = {}): Rule => {
-    return async (tree, context) => {
+    const installDependencies = async (_tree: Tree, context: SchematicContext) => {
+      context.addTask(new AddDevInstall({
+        packageName: moduleToInstall.map((dependency) => `${dependency}${o3rCoreVersion}`).join(' '),
+        hideOutput: false,
+        quiet: false
+      } as any));
+      await lastValueFrom(context.engine.executePostTasks());
+    };
+
+    const updatePackageJson = async (tree: Tree, _context: SchematicContext) => {
```

Comment:
the update of package.json is not done by `AddDevInstall` ?

Reply:
not on the tree